main.go

```diff
@@ -56,6 +57,10 @@ func (a *AWSReadvertiserOptions) validateFlags() error {
 		return fmt.Errorf("The DNS value for the ELB needs to be set properly")
 	}
 
+	if isFQDN := func(domain string) bool {return strings.HasSuffix(domain, ".")}; !isFQDN(a.elb) {
```

Comment:
I see that this tries to be more readable, however the inline closure creation contradicts this effort. I'd suggest to either
* move out `isFQDN` to a separate (probably unexposed) function
* remove the inline closure and just directly use `strings.HasSuffix`